core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -58,6 +58,10 @@ trait MetadataCache {
 
   def hasAliveBroker(brokerId: Int): Boolean
 
+  def isBrokerFenced(brokerId: Int): Boolean
+
+  def isBrokerInControlledShutdown(brokerId: Int): Boolean
```

Comment:
I am not sure that I see the benefits. If we remove those methods, we need to add another one which returns `BrokerRegistration` for a given broker. We would need it in `Partition.isBrokerIsrEligible`. Or did I miss something?

Reply:
I should have been more accurate with my comment. I was thinking that it may be safer to only have these methods in `KRaftMetadataCache`. `isBrokerIsrEligible` would check the type and call those methods if it is right type. We wold also document this complexity in that method: `isBrokerIsrEligible`.

I wanted to protect us from other future code from using these methods by accident.

Reply:
Understood. I think that it is not a bad idea. Let me change this.